MapboxDirections/MBDirectionOptions.swift

```diff
@@ -0,0 +1,454 @@
+        let measurementSystem: MeasurementSystem
+        switch description {
+        case "imperial":
+            measurementSystem = .imperial
+        case "metric":
+            measurementSystem = .metric
+        default:
+            return nil
+        }
+        self.init(rawValue: measurementSystem.rawValue)
+    }
+    
+    public var description: String {
+        switch self {
+        case .imperial:
+            return "imperial"
+        case .metric:
+            return "metric"
+        }
+    }
+}
+
+/**
+ A `RouteShapeFormat` indicates the format of a route’s shape in the raw HTTP response.
+ */
+@objc(MBInstructionFormat)
+public enum InstructionFormat: UInt, CustomStringConvertible {
+    /**
+     The route steps’ instructions are delivered in plain text format.
+     */
+    case text
+    /**
+     The route steps’ instructions are delivered in HTML format.
+     
+     Key phrases are boldfaced.
+     */
+    case html
+    
+    public init?(description: String) {
+        let format: InstructionFormat
+        switch description {
+        case "text":
+            format = .text
+        case "html":
+            format = .html
+        default:
+            return nil
+        }
+        self.init(rawValue: format.rawValue)
+    }
+    
+    public var description: String {
+        switch self {
+        case .text:
+            return "text"
+        case .html:
+            return "html"
+        }
+    }
+}
+
+@objc(MBDirecitonOptions)
+open class DirectionOptions: NSObject, NSSecureCoding, NSCopying {
+    
+    /**
+     Initializes a route options object for routes between the given waypoints and an optional profile identifier.
+     
+     - parameter waypoints: An array of `Waypoint` objects representing locations that the route should visit in chronological order. The array should contain at least two waypoints (the source and destination) and at most 25 waypoints. (Some profiles, such as `MBDirectionsProfileIdentifierAutomobileAvoidingTraffic`, [may have lower limits](https://www.mapbox.com/api-documentation/#directions).)
+     - parameter profileIdentifier: A string specifying the primary mode of transportation for the routes. This parameter, if set, should be set to `MBDirectionsProfileIdentifierAutomobile`, `MBDirectionsProfileIdentifierAutomobileAvoidingTraffic`, `MBDirectionsProfileIdentifierCycling`, or `MBDirectionsProfileIdentifierWalking`. `MBDirectionsProfileIdentifierAutomobile` is used by default.
+     */
+    @objc public init(waypoints: [Waypoint], profileIdentifier: MBDirectionsProfileIdentifier? = nil) {
+        self.waypoints = waypoints
+        self.profileIdentifier = profileIdentifier ?? .automobile
+    }
+    
+    open static var supportsSecureCoding = true
+    
+    // MARK: NSCopying
+    open func copy(with zone: NSZone? = nil) -> Any {
+        let copy = DirectionOptions(waypoints: waypoints, profileIdentifier: profileIdentifier)
```

Comment:
This seems suspect. DirectionsOptions is an abstract class, so why would we create an instance of DirectionsOptions that isn’t a RouteOptions or MatchOptions? `self.init(waypoints:profileIdentifier:)` would seem more appropriate in this case.